hcipy/fourier/chirp_z_transform.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+from scipy.fft import next_fast_len
+
+from .fourier_transform import _get_float_and_complex_dtype
+
+try:
+	import mkl_fft._numpy_fft as _fft_module
+except ImportError:
+	_fft_module = np.fft
+
+class ChirpZTransform:
+	'''The Chirp Z-transform (CZT).
+
+	This class evaluates:
+
+	.. math:: X_k = \sum_{n=0}^{N-1} x(n) z_k^{-n}
+
+	where :math:`N` is the number of points along the real space, :math:`M`
+	is the number of points in the Z-plane. The Chirp Z-transform samples
+	the Z-plane on equidistant points along spiral arcs, where the arcs are
+	given by the starting point :math:`A` and the complex ratio between points
+	:math:`W`:
+
+	.. math:: z_k = A * W^{-k}, k=0, 1, \ldots, M - 1
+
+	This implementation uses Bluestein's algorithm to compute the CZT using Fast
+	Fourier Transforms.
+
+	Parameters
+	----------
+	n : integer
+		The number of points in the real plane.
+	m : integer
+		The number of points in the Z-plane.
+	w : scalar
+		The complex ratio between points in the Z-plane. Note: ensure that its
+		absolute value is close to one, so that w**(max(n, m)**2) does not
+		overflow.
+	a : scalar
+		The starting point in the complex Z-plane. Note: ensure that its absolute
+		value is close to one, so that a**max(n, m) does not overflow.
+	'''
+	def __init__(self, n, m, w, a):
+		# Ensure that w and a are complex scalars.
+		w = complex(w)
+		a = complex(a)
+
```

Comment:
Do we need to add catch values that will cause an overflow error?

Reply:
This is not a trivial check, and imo would be part of the responsibility of the user. Any reasonable implementation that uses chirp Z-transforms would not even come close to overflowing/underflowing the spiral in the Z-plane.

Reply:
Okay. I thought there might be a simple check. I looked through the source code of scipy and they also don't do a check. This issue is resolved.